src/engines/kotor/sound.cpp

```diff
@@ -0,0 +1,98 @@
+/* xoreos - A reimplementation of BioWare's Aurora engine
+ *
+ * xoreos is the legal property of its developers, whose names
+ * can be found in the AUTHORS file distributed with this source
+ * distribution.
+ *
+ * xoreos is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 3
+ * of the License, or (at your option) any later version.
+ *
+ * xoreos is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with xoreos. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/** @file
+ *  An ingame sound in kotor.
+ */
+
+#include "src/aurora/resman.h"
+#include "src/aurora/gff3file.h"
+
+#include "src/sound/sound.h"
+
+#include "src/engines/aurora/util.h"
+
+#include "src/engines/kotor/sound.h"
+
+namespace Engines {
+
+namespace KotOR {
+
+SoundObject::SoundObject(const Aurora::GFF3Struct &sound) {
+	Common::UString temp = sound.getString("TemplateResRef");
+
+	Common::ScopedPtr<Aurora::GFF3File> uts;
+	if (!temp.empty())
+		uts.reset(loadOptionalGFF3(temp, Aurora::kFileTypeUTS, MKTAG('U', 'T', 'S', ' ')));
+
+	const Aurora::GFF3Struct &gff = uts->getTopLevel();
```

Comment:
If you're using `loadOptionalGFF3()`, the return value might be 0! You can't just dereference `uts` here without checking for 0.

Also, `uts` might be 0 if `temp` is empty, of course.

`Common::ScopedPtr::operator->()` does *not* check for 0 itself.

In fact, can anything in this function work if temp is empty at all?